### ISSUE TITLE
Reduce carbon footprint of the website

### ISSUE THREAD

**Issue by KiritoStorm** [zen-browser/www#389]
From the status page tool, I found out there is a [website](https://www.websitecarbon.com) to check how carbon friendly a website is.

The [test](https://www.websitecarbon.com/website/zen-browser-app/) gives me an E rating, which is pretty bad, but acceptible. 

I don't think that this has big priority, but maybe it would make sense to do some optimizations if it doesn't change to much on the website.

It isn't just sustainability, it is also about improving performance.
Less loading time => Less carbon footprint & more performance

Some resources linked to that:
https://beleaf.au
https://sustainablewebdesign.org

**Comment by KyleDL101** [comment]
It seems like the test you checked was outdated by their standards. I did a re-test and it's not dropped to F (we rewrote the backend of the site since it was previously tested).

To be honest, I've tested performance for the site using Google's tools (Lighthouse), and the Zen website gets good performance results.

**Comment by NOCanoa** [comment]
btw @KyleDL101 only thing i can see that can be optimized alot is the fonts
![ss](https://share.thatcanoa.org/ss/zen_Ue1RY8jkCL.png)
2.2mb is kinda alot lol

**Comment by KyleDL101** [comment]
@NOCanoa Agreed, the fonts are the only thing that is concerning.

**Comment by flyri0** [comment]
Hi @KyleDL101,  



I managed to reduce the size of both `JunicodeVF-Regular.woff2` (41.94 KB) and `JunicodeVF-Italic.woff2` (48.76 KB) using subsetting. 
![31122024182412296B](https://github.com/user-attachments/assets/9dfaf8ef-02c7-48f3-9ff0-7440c71c04a8)

It's likely possible to make them even smaller, but that would increase maintenance overhead, as a new subset would need to be generated frequently.  

According to the [Junicode manual](https://github.com/psb1558/Junicode-font/blob/master/docs/JunicodeManual.pdf):  

### 8.1. Subsetting Junicode  

> *First the legalities. It is perfectly all right to create a modified version of Junicode via subsetting, compress it into a webfont (almost certainly in woff2 format), and host it on your web server. This is because “Junicode” is not a “Reserved Font Name” (which complicates web use of many fonts licensed under the Open Font License). If you are nevertheless nervous about the legal requirements of the Open Font License, you can change the font name to something arbitrary with the `--obfuscate-names` option of the `pyftsubset` program, and you can embed the Open Font License, or a link to it, in your CSS. These steps should settle any ambiguity about whether you are in compliance with the license.*  

Let me know if you'd like more details about the subsetting process or any other adjustments!

**Comment by KyleDL101** [comment]
Hi @flyri0, 

That's great! Did you use pyftsubset for this, or something else?

Also, please feel free to open a PR with the subset font files. This is a good change to have.

**Comment by flyri0** [comment]
@KyleDL101 Yes, I used [glyphhanger](https://github.com/zachleat/glyphhanger) to generate a character range based on the main New York Times website (since the website is currently English-only). After that, I used `pyftsubset` to create the subsets and compress them into `woff2`.  

Today, I managed to reduce the size even further (15 KB for each set), but I’ll run a few more tests before opening the PR.

**Comment by KyleDL101** [comment]
@flyri0 Sounds great. Just remember that we do have accent characters in the about us page. Although I think you've already included that but just making sure. Thanks for the work!

**Comment by mauro-balades** [comment]
I have a question. Could we just extract the letters we want from the font file?

**Comment by mauro-balades** [comment]
I also did some changes (A+): https://www.websitecarbon.com/website/zen-browser-app/
